src/amberscript/parser.cc

```diff
@@ -986,8 +992,50 @@ Result Parser::ParseRun() {
     if (!pipeline->IsGraphics())
       return Result("RUN command requires graphics pipeline");
 
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "AS")
+      return Result("missing AS for RUN command");
+
+    token = tokenizer_->NextToken();
+    if (!token->IsString()) {
+      return Result("invalid topology for RUN command: " +
+                    token->ToOriginalString());
+    }
+
+    Topology topo = NameToTopology(token->AsString());
+    if (topo == Topology::kUnknown)
+      return Result("invalid topology for RUN command: " + token->AsString());
+
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "START_IDX")
```

Comment:
This rejects the first DRAW_ARRAY case in the docs:

`RUN <pipeline_name> DRAW_ARRAY AS <topology>`

Reply:
https://github.com/google/amber/issues/475